packages/@lwc/ssr-runtime/src/index.ts

```diff
@@ -173,29 +173,19 @@ export class LightningElement implements PropsAvailableAtConstruction {
     }
 
     getAttribute(attrName: string): string | null {
-        const value = this.__attrs?.[attrName];
-        return value === true ? '' : (value ?? null);
+        return this.__attrs[attrName] ?? null;
     }
 
     setAttribute(attrName: string, value: string | null): void {
-        // Not sure it's correct to initialize here if missing
-        if (!this.__attrs) {
-            this.__attrs = {};
-        }
-
-        if (value === null) {
-            delete this.__attrs[attrName];
-        } else {
-            this.__attrs[attrName] = value;
-        }
+        this.__attrs[attrName] = String(value);
     }
 
     hasAttribute(attrName: string): boolean {
         return Boolean(this.__attrs && attrName in this.__attrs);
     }
 
     removeAttribute(attrName: string): void {
-        delete this.__attrs?.[attrName];
+        this.__attrs[attrName] = null;
```

Comment:
This seems unnecessary to me. `getAttribute` already converts to `null` if it's not in the property bag.